### PR TITLE
flash_tool.py: Return if there is no board connected

### PR DESCRIPTION
**Pull request by oliviermartin**
Prevent this tool to block if there is no board connected.

### CODE REVIEW

**Comment by c1728p9** [comment]
Thanks for the pull request.  Instead of making this a command line option, it might be better just to make it always non-blocking.  Not sure if anyone else has preferences on this, but I don't see much of a use in having flash tool block.

**Comment by 0xc0170** [comment]
>  it might be better just to make it always non-blocking

What do you think @oliviermartin ? You can always write blocking on top of it.

**Comment by oliviermartin** [comment]
For me making the tool non blocking it is even better because I do not need to change my script that invokes this tool!

**Comment by 0xc0170** [comment]
LGTM

File: pyOCD/tools/flash_tool.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def main():
     else:
         board_selected = MbedBoard.chooseBoard(board_id=args.board_id, target_override=args.target_override,
```
Comment (c1728p9) [inline pyOCD/tools/flash_tool.py]: I think chooseBoard is blocking by default, so you'll need to set the 'blocking' parameter to false.

**Comment by c1728p9** [comment]
Hi @oliviermartin , have you had a chance to look at my comment?

File: pyOCD/tools/flash_tool.py (hunk 1 above)
Comment (oliviermartin) [inline pyOCD/tools/flash_tool.py]: You are right, I accidentally removed this change in my previous update. The new version should restore the initial parameter setting.

**Comment by c1728p9** [comment]
test this please

**Comment by janjongboom** [comment]
+1 but I want an error message too

**Comment by c1728p9** [comment]
Hi @oliviermartin , could you rebase this commit on top of master and add an error message?  Once it is rebased I'll run CI testing on it and merge it.

**Comment by c1728p9** [comment]
test this please

**Comment by oliviermartin** [comment]
Tested-By: Olivier Martin <olivier.martin@labapart.com>

With no board connected:

```
$ ./flash_tool.py 
No connected boards
Error: There is no board connected.
```

With KL25Z connected:

```
$ ./flash_tool.py -l
0 => FRDM-KL25Z [kl25z] boardId => 02000203D62D2E6C2BD1D394
```

With KL25Z connected:

```
$ ./flash_tool.py -t kl25z -ce
INFO:root:DAP SWD MODE initialised
INFO:root:KL25Z not in secure state
INFO:root:2 hardware breakpoints, 0 literal comparators
INFO:root:CPU core is Cortex-M0+
INFO:root:2 hardware watchpoints
Erasing chip...
Done
```

With no board connected:

```
$ ./flash_tool.py -t kl25z -ce
No connected boards
Error: There is no board connected.
```

**Comment by c1728p9** [comment]
The phrase "test this please"  triggers the CI system to run tests.  Wanted to let you know so it doesn't seem like I'm being demanding asking you to run tests :smile:.Thanks for the testing though.

**Comment by c1728p9** [comment]
Thanks for the PR @oliviermartin

**Pull request merged**